### DISCUSSION TITLE
Should we release 0.72.4?

### DISCUSSION THREAD

**Discussion by kelset** [reactwg/react-native-releases#83] [Patches]
## Released, please refer to #85 

---

Current Release: [0.72.3](https://github.com/facebook/react-native/releases/tag/v0.72.2)

Conversations on this thread are limited:

- [major release issues](https://reactnative.dev/contributing/release-faq#what-is-release-blocking).
- [qualified cherry-pick requests](https://reactnative.dev/contributing/release-faq#what-is-a-qualified-pick-request) of commits on main that [did not make the previous patch version](https://reactnative.dev/contributing/release-faq#how-do-i-know-if-my-fixfeature-is-in-a-certain-release).

Please include a link to the specific commit on main to be cherry-picked, for example: [facebook/react-native@bd2b7d6](https://github.com/facebook/react-native/commit/20b0eba581a00e5e7e300f6377379b836617c147)

In other words, if you cannot point to a particular commit on main, then your request likely belongs as a new issue.
If the issue is a [major release issues](https://reactnative.dev/contributing/release-faq#what-is-release-blocking), please reference the issue here.

---

## No more pick requests are being accepted unless exceptional.
 Release crew is waiting on Metro changes and then will do testing

#### Commits Picked
* `enterKeyHint` types on TextInput
  * https://github.com/facebook/react-native/commit/5062553c62ea743e7579ef57aceeec8636121c99 	
* Allow string transform style in TypeScript 
  * https://github.com/facebook/react-native/commit/2558c3d4f56776699602b116aff8c22b8bfa176a
* Support for Android 14
  * For targeting SDK 34 - Added RECEIVER_EXPORTED/RECEIVER_NOT_EXPORTED: https://github.com/facebook/react-native/commit/177d97d8ea962bdd4dad8fcf0efb04a307f25000
* Merged https://github.com/facebook/react-native/pull/38785
* Merged https://github.com/facebook/react-native/pull/38778
* Fix inverted flatlist android 13 https://github.com/facebook/react-native/commit/6d206a3f54725f7f53692222293a9d1e58b11ca4 & https://github.com/facebook/react-native/commit/3dd816c6b7bcd9cc4c21199f0b645755fb97f50f
* Fabric Interop
  * https://github.com/facebook/react-native/commit/995bb3e995e47c389db421d5658d3336053e0d10
  * https://github.com/facebook/react-native/commit/ccc50ddd2d1b548ddd72e15f39fcb8407ee7963b
* Fix Android ScrollView not responding to Keyboard events when nested inside a KeyboardAvoidingView
  * https://github.com/facebook/react-native/commit/c616148a05c00728b80d2fd9dcbd6f15d08a2dfa
* fix: Correctly assign the hermes-engine pod tag when installing pod
  * https://github.com/facebook/react-native/commit/08688d3b9f05b790e1af74b013552ff1789618b8
* Remove option to paste rich text from Android EditText context menu
  * https://github.com/facebook/react-native/commit/b1ceea456d1cdc00c723582d00e5ae585f066b55
* Released a new Hermes version with pick:
  * https://github.com/facebook/hermes/commit/8fd20f29af83ab59c5e21d7de8212bcc904dcc4a
* ~~Update new_architecture.rb for React-ImageManager~~
  * ~~https://github.com/facebook/react-native/commit/1761aea17551dbcee62fecb210497ec3f9f1b7c3~~ dropped 
* Restore checking shadow tree commit cancellation after commit hook 
  * https://github.com/facebook/react-native/commit/a30f3938a5c9ad8becb9dd509c71cb328ac09026
* Pass hitSlop prop into TextInput Pressability config 
  * https://github.com/facebook/react-native/commit/d6ff5c6d5b73518d506c45b3d8519e66d2975243 

 
#### Waiting on
* metro-side: @huntie is taking care of this
  * Merged in Metro https://github.com/facebook/metro/commit/692293339e3b9f542f7b83c04bdccf46fef05766
  * Merged in CLI https://github.com/react-native-community/cli/pull/2048
  * Committed as https://github.com/facebook/react-native/pull/38898 by @lunaleaps 

#### Will not pick
* https://github.com/facebook/react-native/pull/35803

#### List of issues we're aware of
* https://github.com/facebook/react-native/issues/37871
* https://github.com/facebook/react-native/issues/38204

#### Local commits to backport to main
1. https://github.com/facebook/react-native/commit/8ce471e2fa802cc50ff2d6ab346627cb5f6d79b4 (from 72.1)
2. https://github.com/facebook/react-native/commit/8f41f25c214f995073e90b786c805eb45ff7dee5 (TBD if this revert will be done in main branch too, depends on [this discussion](https://github.com/react-native-community/discussions-and-proposals/discussions/687) and the [planning for 0.73](https://github.com/reactwg/react-native-releases/discussions/64) )

**Comment by efstathiosntonas** [comment]
`TextInput`/`Text` party, newest PR to oldest:

1. https://github.com/facebook/react-native/pull/38306
2. https://github.com/facebook/react-native/pull/38293
3. https://github.com/facebook/react-native/pull/38258
4. https://github.com/facebook/react-native/pull/38210
5. https://github.com/facebook/react-native/pull/38123
6. https://github.com/facebook/react-native/pull/38110
7. https://github.com/facebook/react-native/pull/38014
8. https://github.com/facebook/react-native/pull/37998
9. https://github.com/facebook/react-native/pull/37766
10. https://github.com/facebook/react-native/pull/37271
11. https://github.com/facebook/react-native/pull/37147
12. https://github.com/facebook/react-native/pull/36979
13. https://github.com/facebook/react-native/pull/36586
14. https://github.com/facebook/react-native/pull/36436
15. https://github.com/facebook/react-native/pull/36428
16. https://github.com/facebook/react-native/pull/35908
17. https://github.com/facebook/react-native/pull/35878
18. https://github.com/facebook/react-native/pull/35130
19. https://github.com/facebook/react-native/pull/35704
20. https://github.com/facebook/react-native/pull/35703
21. https://github.com/facebook/react-native/pull/34645
22. https://github.com/facebook/react-native/pull/31376
23. https://github.com/facebook/react-native/pull/30995
24. https://github.com/facebook/react-native/pull/29942
25. https://github.com/facebook/react-native/pull/29847
26. https://github.com/facebook/react-native/pull/37271
27. https://github.com/facebook/react-native/pull/37424

**Reply by kelset** [reply]
that's a massive list, I'm not gonna lie - and none are merged 😅 can you give a bit more context? is there some common thread going through all of them?

**Reply by efstathiosntonas** [reply]
@kelset some of them are open for quite some time and I feel that the textinput/text components haven’t got so much attention in the latest releases. Some of them are giving developers headaches to workaround. This list is more like “they need some attention and love”, I do not expect all of them to be included in a single/patch release. 

If the comment goes too far then let’s mark it as offtopic.

**Reply by efstathiosntonas** [reply]
Added this on the list

- https://github.com/facebook/react-native/pull/37271  

Fixes this: https://github.com/facebook/react-native/issues/15114

**Reply by kelset** [reply]
gotcha - yeah honestly I understand your POV in trying to shine a light on this; I wish I had a good explanation or a plan we could act on about it, but it's out of my hands 😓

Probably this is not the right place to have this conversation but until someone from Meta chimes in let's leave it open

**Reply by efstathiosntonas** [reply]
Added to the list:

- https://github.com/facebook/react-native/pull/37424

**Reply by cortinico** [reply]
Hey @efstathiosntonas - Nicola here from the React Native team. 

We do try to review and merge as many PRs as we could, but sometimes things fall through the crack as PRs were old (opened before we had a review process in place) or just more complex and require time to review.

Thanks for sharing this list, but it's harder for us to merge them all at once. What would be the most crucial one (and why)? We can prioritise them and try to get them merged.

For the rest, these releases discussions should be kept for standard cherry-pick requests; so I'll go ahead and hide this as "off topic". But thanks for forcing the conversation to happen.

**Reply by efstathiosntonas** [reply]
@cortinico Hi,  I really can't tell which one should be reviewed, imho they are all important since they do not have workaround fixes. I guess you should pick the ones that are the easiest to review 🤷‍♂️ . The comment is not marked as off-topic (yet?), just fyi. Thank you for all your hard work guys/gals.

**Comment by kelset** [comment]
Need a CLI bump: https://github.com/facebook/react-native/issues/38270#issuecomment-1633802996

11.3.6

**Reply by szymonrybczak** [reply]
https://github.com/facebook/react-native/pull/38778

**Reply by lunaleaps** [reply]
picked

**Comment by cortinico** [comment]
This is a fix for the Interop Layer to support `react-native-maps` and other libraries that are using direct events:
- https://github.com/facebook/react-native/pull/38352

Not merged yet, but should be picked as soon as it's out.

**Reply by cortinico** [reply]
It was merged so should be good to be picked up

**Reply by lunaleaps** [reply]
picked

**Comment by mwaleed-39** [comment]
I faced some issue in react native. when we pressed upload image, the error message shows:

How to call this API, when we upload image in front end side, the image will be stored in mongodb
I create API:
[app.post](http://app.post/)('/finerequest', (req, res, next) => {

const file = req.files.image;
cloudinary.uploader.upload(file.tempFilePath, (err, result) => {
console.log(result)
const requestId = req.body.request_id
const fine = new Fine({
_id: new mongoose.Types.ObjectId,
request_id: requestId,
image: result.url
})
fine.save()
.then(result => {
res.status(200).json({
message: "Fine and Vehicle Repair Request added",
response: result
})
})
.catch(err => {
res.status(500).json({
message: "Fine and Vehicle request not added",
error: err
})
})
})
})

CODE:

import React, { useState } from 'react'
import { Alert, Button, View,StyleSheet } from 'react-native'
import * as ImagePicker from 'react-native-image-picker'
import {launchImageLibrary} from 'react-native-image-picker'

const options ={
title: 'Select Image',
type: 'library',
options: {
maxHeight:200,
maxWidth:200,
selectionLimit: 1,
mediaType: 'photo',
includeBase64: false,
}
}

const Request = () => {
const opengallery = async()=>{
const image = await ImagePicker.launchImageLibrary(options);
console.log(image)
}

return (

)
}
export default Request;

const styles = StyleSheet.create({
container:{
flex:1,
alignItems:'center',
justifyContent:'center'
}
})

when we pressed upload button, the error message shows:

Possible Unhandled Promise Rejection (id: 1):
TypeError: Cannot read property 'launchImageLibrary' of null
TypeError: Cannot read property 'launchImageLibrary' of null

Again and again same error show, we Image picker library, I think it has been deprecated. Provide Solution


Image pIcker or launchImageLibrary are not working

**Reply by fortmarek** [reply]
Please, create a new issue in the `react-native` repository instead, including with reproduction steps. This discussion is only meant for pick requests of commits already in `main`.

**Comment by mwaleed-39** [comment]
import React, { useState, useEffect } from 'react';
import { Button, Image, View, Platform,StyleSheet } from 'react-native';
import * as ImagePicker from 'expo-image-picker';
import { useRoute } from ***@***.***/native';
export default function Fine_Repair_Request() {
  const requestId = useRoute().params?.request_id;
  const [image, setImage] = useState(null);

  const pickImage = async () => {
    // No permissions request is necessary for launching the image library
    let result = await ImagePicker.launchImageLibraryAsync({
      mediaTypes: ImagePicker.MediaTypeOptions.All,
      allowsEditing: true,
      aspect: [4, 3],
      quality: 1,
    });

    console.log(result);

    if (!result.canceled) {
      setImage(result.assets[0].uri);
    }
  };


  const handleSubmit = async () => {

    if (image) {
      const formData = new FormData();
      formData.append('image', {
        uri: image,
        type: 'image/jpeg', // Adjust the image type if needed
        name: 'image.jpg', // Adjust the filename if needed
      });
      formData.append('request_id', requestId);
      try {
        const response = await fetch('http://192.168.10.1:5000/finerequest',
{
          method: 'POST',
          body: formData,
          headers: {
            'Content-Type': 'multipart/form-data',
          },
        });

        const data = await response.json();
        console.log(data); // Handle the response data if needed
      } catch (error) {
        console.error('Error submitting the image:', error);
      }
    }
  };

  return (
    <View style={{ flex: 1, alignItems: 'center', justifyContent:
'center',marginBottom:50 }}>
      <Button title="Upload Image" onPress={pickImage} style
={styles.button}  />
      {image && <Image source={{ uri: image }} style={{
marginBottom:20,width: 200, height: 200 }} />}
      <Button title="Submit" onPress={handleSubmit} />
    </View>
  );
}
const styles = StyleSheet.create({
  button:{
    marginBottom:50
  }
})

Error: Key "cancelled" in the image picker result is deprecated and will be
removed in SDK 48, use "canceled" instead

**Reply by efstathiosntonas** [reply]
@mwaleed-39 stop spamming this discussion, this is not stackoverflow, no one will help you. Follow these instructions: https://github.com/reactwg/react-native-releases/discussions/83#discussioncomment-6464811

**Comment by t0nyh0** [comment]
Can we consider this for apps implementing chat-style interfaces:

https://github.com/facebook/react-native/commit/6d206a3f54725f7f53692222293a9d1e58b11ca4

**Reply by liamjones** [reply]
To finish off the workaround, you will also need https://github.com/facebook/react-native/pull/38073 (not yet merged).

**Reply by Titozzz** [reply]
I think we'll wait for both to landed to cherry pick :)

**Reply by t0nyh0** [reply]
Looks like this has been merged.  Can we cherry pick? :-)

**Reply by Titozzz** [reply]
added both to the list

**Reply by hannojg** [reply]
Specifically its these two PRs: 

- https://github.com/facebook/react-native/pull/38071
- https://github.com/facebook/react-native/pull/38073

**Reply by lunaleaps** [reply]
picked

**Comment by cortinico** [comment]
There is another fix for Fabric Interop for Android that extends the set of libraries we support:

- https://github.com/facebook/react-native/pull/38527

To be picked once it's merged

**Reply by lunaleaps** [reply]
I'm getting a pick conflict on this. Asking @cortinico to help merge: https://github.com/facebook/react-native/pull/38823/files

**Reply by cortinico** [reply]
Here the resolution of the conflicts:
- https://github.com/facebook/react-native/pull/38835

**Comment by nvodoor** [comment]
Could we consider cherry-picking in https://github.com/facebook/react-native/commit/177d97d8ea962bdd4dad8fcf0efb04a307f25000? It is needed for Android 14 support.

**Reply by lunaleaps** [reply]
picked

**Comment by billnbell** [comment]
Can we consider https://github.com/facebook/react-native/pull/38247 it added React-ImageManager

**Reply by lunaleaps** [reply]
Can you expand on why this is necessary for this patch? Is it fixing essential behavior?

**Reply by billnbell** [reply]
I am pretty sure React-ImageManager was missed and should be in the list of core modules. It is needed for several react-native-blog* and fast-image, etc. So for me it is critical and needed, plus should have been there from 0.72.0

**Reply by fortmarek** [reply]
After discussing it with the release crew, we decided to pick this one (already done). Thanks for raising this!

**Reply by lunaleaps** [reply]
We've had to drop this commit because it has conflicts with the release branch. It's been removed from 0.72 for now

**Comment by fabOnReact** [comment]
Could we consider `Remove option to paste rich text from Android EditText context menu` https://github.com/facebook/react-native/commit/b1ceea456d1cdc00c723582d00e5ae585f066b55? Fixes an issue reported in several chat applications ([Expensify](https://github.com/Expensify/App/issues/21411) and [Zulip-Mobile](https://github.com/zulip/zulip-mobile/issues/4660#issuecomment-951338651)). Thanks a lot.

**Reply by lunaleaps** [reply]
picked

**Comment by efstathiosntonas** [comment]
Please consider picking this, it solves an issue with KeyboardAvoidingView and ScrollView on Android: 
- https://github.com/facebook/react-native/pull/38728

**Reply by lunaleaps** [reply]
picked

**Comment by hannojg** [comment]


**Reply by hannojg** [reply]
Whoops, has already been raised here: https://github.com/reactwg/react-native-releases/discussions/83#discussioncomment-6497407

**Reply by fortmarek** [reply]
Thanks, yes, we will try picking both of these!

**Comment by Titozzz** [comment]
https://github.com/facebook/react-native/commit/5062553c62ea743e7579ef57aceeec8636121c99 adds missing types to a prop that has been added in 0.71

**Reply by lunaleaps** [reply]
picked

**Comment by cipolleschi** [comment]
https://github.com/facebook/react-native/pull/38754 this is landed today, but it will solve an update problem with hermes, so it would be nice to have it in 0.72.4.

**Reply by lunaleaps** [reply]
picked

**Comment by efstathiosntonas** [comment]
Hey folks, is there an eta for 0.72.4? Thanks

**Comment by NickGerleman** [comment]
Typings fix for string transform style: https://github.com/facebook/react-native/commit/2558c3d4f56776699602b116aff8c22b8bfa176a

**Reply by lunaleaps** [reply]
picked

**Comment by emzet93** [comment]
Performance improvement for `JSON.stringify`: https://github.com/facebook/hermes/commit/8fd20f29af83ab59c5e21d7de8212bcc904dcc4a

It's related to https://github.com/facebook/hermes/issues/1008 and brings `JSON.stringify` performance closer to JSC implementation

**Reply by lunaleaps** [reply]
Asking the release crew whether this should be part of the pick

**Reply by lunaleaps** [reply]
picked

**Comment by billnbell** [comment]
ETA?

**Reply by cipolleschi** [reply]
Please don't ask ETAs in this discussion. This is meant to be used only to coordinate the release.
The release crew will release a new version of the framework when it's ready.

**Comment by Kudo** [comment]
since xcode 15 is probably coming before 0.73, it would be great to backport the workaround to 0.72: https://github.com/facebook/react-native/commit/10d55888cce0b625c4b05a19ed609154893d17d0

**Reply by lunaleaps** [reply]
I think this has already been picked to 0.72 https://github.com/facebook/react-native/blob/0.72-stable/packages/react-native/scripts/cocoapods/__tests__/utils-test.rb#L438

**Reply by Kudo** [reply]
oh my bad. didn't notice that is included from 0.72.0. that's awesome. thanks!

**Comment by efstathiosntonas** [comment]
Can we pick this:
- https://github.com/facebook/react-native/pull/38715

it will improve reanimated performance and fix some crashes from both sides: https://github.com/software-mansion/react-native-reanimated/pull/4833

**Reply by lunaleaps** [reply]
picked

**Comment by lunaleaps** [comment]
We are not taking anymore pick requests. We need to test at this stage.

**Comment by efstathiosntonas** [comment]
can we pick this: 🙏 
- https://github.com/facebook/react-native/pull/38857

@lunaleaps sorry for rising this now but it's a really important prop

**Reply by lunaleaps** [reply]
picked

**Comment by kelset** [comment]
release crew specific improvement to merge asap (after patch is ok) -> https://github.com/facebook/react-native/pull/38888

doesn't need a new patch, but merge for consistency across all versions in support window + main

**Comment by ravindraguptacapgemini** [comment]
@kelset Can you please update regarding the status of the release?
